ion/unmarshal.go

```diff
@@ -0,0 +1,673 @@
+package ion
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+var (
+	// ErrNoInput is returned when there is no input to decode
+	ErrNoInput = errors.New("ion: no input to decode")
+)
+
+// Unmarshal unmarshals Ion data to the given object.
+func Unmarshal(data []byte, v interface{}) error {
+	return NewDecoder(NewReader(bytes.NewReader(data))).DecodeTo(v)
+}
+
+// UnmarshalStr unmarshals Ion data from a string to the given object.
+func UnmarshalStr(data string, v interface{}) error {
```

Comment:
Not really necessary and adds another function to the `Unmarshal` story.

Reply:
#29